include/cudaq/Optimizer/Builder/Intrinsics.h

```diff
@@ -55,6 +55,10 @@ static constexpr const char createCudaqStateFromDataFP32[] =
 // Delete a state created by the runtime functions above.
 static constexpr const char deleteCudaqState[] = "__nvqpp_cudaq_state_delete";
 
+// Get state of a kernel (placeholder function, calls are always replaced in
+// opts)
+static constexpr const char getCudaqState[] = "__nvqpp_cudaq_state_get";
```

Comment:
Do you want to stick with a dummy function or would you rather introduce a NOP "conversion" Op to the IR for this?

Reply:
I am fine either way. An OP gives an advantage of not having to add the declaration to the MLIR for this function and having to remove it in StateValidation pass (but there are more to remove so StateValidation is still needed).

Reply:
Actually, I was thinking about adding ops for a few runtime functions related to state, so it would be easier to remove those ops in optimizations and only add the runtime functions if needed. I would like to do that in a separate PR, for all the state-related runtime functions:

```cpp
// The internal data of the cudaq::state object must be `2**n` in length. This
// function returns the value `n`.
static constexpr const char getNumQubitsFromCudaqState[] =
    "__nvqpp_cudaq_state_numberOfQubits";

// Create a new state from data.
static constexpr const char createCudaqStateFromDataFP64[] =
    "__nvqpp_cudaq_state_createFromData_fp64";
static constexpr const char createCudaqStateFromDataFP32[] =
    "__nvqpp_cudaq_state_createFromData_fp32";

// Get state of a kernel (placeholder function, calls are always replaced in
// opts)
static constexpr const char getCudaqState[] = "__nvqpp_cudaq_state_get";
```

For the first 3 functions, we would still have the runtime intrinsics defined but calls to them added only if the corresponding op is not removed by opts. The last function would not be needed as the corresponding op is always removed by the opts. WDYT?

Reply:
Replaced `"__nvqpp_cudaq_state_get"` intrinsic by `cc.get_state` op, this simplified the code quite a bit. I will create a follow-up PR for the rest of the state-related op additions, as it is not related to the current change.